### PR TITLE
[WIP] Remove old artifact code

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary
```SUMMARY: Infrastructure "Remove old artifact code"```

#### Purpose of change
The old artifact code was bad in a couple of ways:
1. hardcoded. there was a ton of data that was hardcoded in that made all of the following points even worse. It is true that this code existed in the time of whales, which means there wasn't any json loading, but it is time to forge a new path.
2. not easily extendable - it was basically impossible to extend this code to add new effects without adding a ton of data and without C++ knowledge. `relic`, which is the nouveau artifact class, has gone the extra step to make sure that as much of it as possible is able to be written into json.
3. tip-toed around Item_factory. this code created new itypes from hardcode in order for the artifacts to work, and saved them in artifacts.gsav in order for the item to be saved properly. It is 100% possible to simply use `itype`s from json and then save the relic_data attached to the `item` without having to save newly created itypes.
4. overused enums. Practically all of the enums could have been done in a rolling fashion, which is what i did in relic. artifact_property stays because it has some cool mapgen effects, but STR_UP giving a hard +3 to strength has been replaced with a map of mod::value to actual number. Additionally, the replacement for AEA are spells. 
5. hard to follow. this might have been a me thing, and possibly is compounded with all the data encoded in the header file, but all the classes inheriting from itype made it pretty confusing as to what artifacts were, and made it hard to extend.

These are some of the things that old artifacts did that the new ones do not, and that i (or someone in the dev team _wink_) plan to reimplement in the near future:
1. custom/random artifact names. old artifact names had names generated from what type of artifact effects they had.
2. activated effects. when you activate an artifact, it did its effect. this will be replaced with spells, and using up the charges on the relic_data (done)
3. recharging methods. there were various recharging methods for artifacts that allowed you to continue to use the artifact you found. (done)
#### Describe the solution
delete artifact.h and artifact.cpp and deal with the fallout. also delete the artifact enums in enum.h (except artifact_natural_property which stays for some mapgen stuff)

#### Testing
Loaded the game, first with vanilla, then magiclysm, then aftershock. i didn't see any debug messages but i'm pretty sure there might be some unvisited members in json somewhere now. i'll just have to check travis's output.

#### Additional context
In my opinion the artifact code was the worst part of our codebase. I count deleting it all as a huge win, and its replacement already far outshines it even with the features that are currently incomplete.

Requires #42199 and #42137 and has the commits from those PRs included.

### CODE REVIEW

**Comment by Axema** [comment]
What other artifact properties and charge methods will need to be recreated, out of curiosity? I recall AEP_EXTINGUISH, fairly certain most can now be easily done via enchantments? What about transforming relics? I have active work in my projects that require the flexibility artifacts have, and I know others have talked about it before given Foulman's bug report on it.

**Comment by KorGgenT** [comment]
If you found the old artifacts flexible, I think you will be very pleasantly surprised with relics. I don't think I need to go into the details beyond linking the relevant documentation on that front.

I am not the one doing the initial implementation of artifact activations, but i'm fairly sure the initial charge type will be over time. I am a big fan of iterative feature implementation, so additional charging types shall be implemented over time as needed/wanted/time allows.

The transformation of items into other items changing their relic status is known to me, but I neither consider it a bug nor have come up with a reasonable plan for implementing an analogue at this time.

**Comment by KorGgenT** [comment]
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/MAGIC.md Forgot to link the doc, it's at the bottom under "enchantments"

**Comment by Axema** [comment]
> nor have come up with a reasonable plan for implementing an analogue at this time.
Enchantments can have conditions, right? Would checking for "is the base item currently active" and making it a valid condition to check for be easier than trying to implement a way to make item transform change relic data outright?

**Comment by KorGgenT** [comment]
certainly don't have any idea what's going on with the vision tests.

File: src/character.cpp [hunk 1]
```diff
@@ -6873,7 +6848,7 @@ float Character::active_light() const
 
     if( lumination < 60 && has_active_bionic( bio_flashlight ) ) {
         lumination = 60;
-    } else if( lumination < 25 && has_artifact_with( AEP_GLOW ) ) {
+    } else if( lumination < 25 ) {
         lumination = 25;
```
Comment (kevingranade) [inline src/character.cpp:6852]:
```suggestion
```

**Comment by kevingranade** [comment]
Found it, slight error when removing a light manipulation artifact effect.

**Comment by kevingranade** [comment]
Need to remove dead artifact_data json fields.

**Comment by KorGgenT** [comment]
all right, will do.

**Comment by KorGgenT** [comment]
NOTE: can't _quite_ delete the artifact code yet, as aftershock uses artifact recharging mechanics which are still WIP for relic. so, i'm going to convert this to a draft for the time being.

**Comment by KorGgenT** [comment]
image
